llvm/test/CodeGen/AIE/aie2/hardware-loops/zol-md-accept-reject.ll

```diff
@@ -0,0 +1,79 @@
+; This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+; See https://llvm.org/LICENSE.txt for license information.
+; SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+;
+; (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+
+; RUN: llc -O2 -stop-after=hardware-loops -mtriple=aie2 --enable-aie-hardware-loops \
+; RUN:    --enable-aie-zero-overhead-loops %s -o - | FileCheck %s
+
+define void @simple_loop_accept_no_md(i32 noundef %n, ptr nocapture readonly %in, ptr nocapture writeonly %out) {
+entry:
+  %cmp4 = icmp sgt i32 %n, 0
+  br i1 %cmp4, label %for.body, label %for.cond.cleanup
+
+for.cond.cleanup:                                 ; preds = %for.body, %entry
+  ret void
+
+; CHECK: [[DEC:%[0-9]+]] = call i1 @llvm.loop.decrement.i32(i32 1)
```

Comment:
Shouldn't `no metadata` be counted as `mintripcount == 0`?

Reply:
Maybe it is not necessary. With no metadata we will accept as HL but will reject for SWP. I think this type of loop can be part of the initialization, for example. In this case we can let to the compiler to unroll and apply HL freely. What do you think?

Reply:
Looking again and benchmarking again I see this as a good think, it is included.

Reply:
I'm not quite sure I know what you mean. DownCountLoop would detect a static tripcount if it was given as a constant, even if no metadata was present. I don't think that's much more difficult for ZOL, without 'annotating' it, if with that you mean providing min-itercount metadata from the value of the constant.
So, in zol construction,  if you have a constant tripcount, I would allow it as a hardware loop, trusting that SWP would find the same tripcount from the LoopStart constant.